File: src/console/command/impl/HelpCommand.php
<?php

namespace pocketcloud\cloud\console\command\impl;

use pocketcloud\cloud\console\command\parameter\def\CommandNameParameter;
use pocketcloud\cloud\console\command\Command;
use pocketcloud\cloud\console\command\CommandManager;
use pocketcloud\cloud\console\command\sender\ICommandSender;
use pocketcloud\cloud\console\command\SubCommand;

final class HelpCommand extends Command {

    public function __construct() {
        parent::__construct("help", "List all commands", ["?"]);
        $this->addParameter(new CommandNameParameter(
            "command",
            true
        ));
    }

    public function run(ICommandSender $sender, string $label, array $args, ?SubCommand $subCommand = null): bool {
        $originCommand = $args["command"] ?? null;
        $commands = $originCommand === null ? CommandManager::getInstance()->getAll() : [$originCommand];

        foreach ($commands as $command) {
            $usages = explode("\n", $command->getUsage());
            $extraPrint = count($usages) > 1;
            $sender->info("§b" . $command->getName() .
                " §8- §r" . $command->getDescription() .
                ($extraPrint || $originCommand === null ? "" : " §8- §r" . current($usages)) .
                " §8- §r[§c" . (empty($command->getAliases()) ? "" : implode("§8, §c", $command->getAliases())) . "§r]"
            );

            if ($extraPrint && $originCommand !== null) {
                foreach ($usages as $usage) {
                    $sender->info("§8» §r" . $usage);
                }
            }
        }
        return true;
    }
}

File: cloud-core/src/main/java/de/pocketcloud/cloud/console/command/impl/HelpCommand.java
package de.pocketcloud.cloud.console.command.impl;

import de.pocketcloud.cloud.PocketCloud;
import de.pocketcloud.cloud.console.command.Command;
import de.pocketcloud.cloud.console.command.ctx.CommandContext;
import de.pocketcloud.cloud.console.command.desc.CommandDescription;
import de.pocketcloud.cloud.console.command.parameter.def.LiteralCommandParameter;
import de.pocketcloud.cloud.console.command.sender.CommandSender;

import java.util.ArrayList;
import java.util.List;

@CommandDescription(name = "help", description = "List all commands", aliases = {"?"})
public final class HelpCommand extends Command {

    @Override
    public void prepare() {
        addParameter(new LiteralCommandParameter("command", true, "Command not found"));
    }

    @Override
    public boolean run(CommandSender sender, CommandContext ctx) {
        List<Command> commands = new ArrayList<>();
        if (ctx.hasArg("command")) {
            commands.add(ctx.argCommand("command"));
        } else {
            commands.addAll(PocketCloud.instance().commandManager().getAll());
        }

        sender.info("Commands §8(§b{}§8)§r:", commands.size());
        for (Command command : commands) {
            if (command.getAliases().length > 0) {
                sender.info("§b{} §8- §r{} §8- §r[§c{}§r]", command.getName(), command.getDescription(), String.join("§8, §c", command.getAliases()));
            } else {
                sender.info("§b{} §8- §r{} §8- §r[]", command.getName(), command.getDescription());
            }
        }

        return true;
    }
}